vnext/Microsoft.ReactNative/IViewManager.idl

```diff
@@ -103,4 +103,46 @@ namespace Microsoft.ReactNative
     )
     void OnDropViewInstance(XAML_NAMESPACE.FrameworkElement view);
   };
+
+  [webhosthidden]
+  [experimental]
+  DOC_STRING(
+      "An experimental API. It may be removed or changed in a future version. "
+      "Used by @IViewManagerWithNativeLayout for the measurement callback measure mode parameter."
+  )
+  enum YogaMeasureMode {
+    Undefined,
+    Exactly,
+    AtMost,
+  };
+
+  [webhosthidden]
+  [experimental]
+  DOC_STRING(
+      "An experimental API. It may be removed or changed in a future version. "
+      "Used by @IViewManagerWithNativeLayout for the measurement callback return value."
+  )
+  struct YogaSize {
+    Single Width;
+    Single Height;
+  };
+
+  [webhosthidden]
+  [experimental]
+  DOC_STRING(
+    "An experimental API. It may be removed or changed in a future version. "
+    "Enables view managers to implement custom Yoga leaf node measurement."
+  )
+  interface IViewManagerWithNativeLayout {
```

Comment:
should this supersede IViewManagerRequiresNativeLayout ?

Reply:
I think probably not. RequiresNativeLayout has some other implicit behavior like not setting Yoga props on children here:
https://github.com/microsoft/react-native-windows/blob/f865871c49fe66f5e5b0072764bc5bdb75b3a21b/vnext/Microsoft.ReactNative/Views/ViewManagerBase.cpp#L341

For this version of the interface, I'd strictly like to avoid skipping layout, because I want to use this to allow a control to self measure / self arrange, but then allow recursive Yoga layout on children.